File: background-geofence/src/ios/background-geofence.m
/********* background-geofence.m Cordova Plugin Implementation *******/

#import <Cordova/CDV.h>

@interface background-geofence : CDVPlugin {
  // Member variables go here.
}

- (void)coolMethod:(CDVInvokedUrlCommand*)command;
@end

@implementation background-geofence

- (void)coolMethod:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
    NSString* echo = [command.arguments objectAtIndex:0];

    if (echo != nil && [echo length] > 0) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
    }

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end

File: BackgroundGeofence/src/ios/BackgroundGeofence.swift
@objc(BackgroundGeofence) class BackgroundGeofence : CDVPlugin {
  @objc(monitorGeofence:)
  func monitorGeofence(command: CDVInvokedUrlCommand) {
    var pluginResult = CDVPluginResult(
      status: CDVCommandStatus_ERROR
    )

    let coordinates = command.arguments[0] as? String ?? ""
    //let coordinates = command.arguments[0] as? Dictionary ?? [longitud:0, lattidude: 0]

    let toastController: UIAlertController =
      UIAlertController(
        title: "Plugin",
        message: "In the plugin",
        preferredStyle: .alert
      )
      
    self.viewController?.present(
      toastController,
      animated: true,
      completion: nil
    )

    DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
        toastController.dismiss(
            animated: true,
            completion: nil
        )
    }

    pluginResult = CDVPluginResult(
    status: CDVCommandStatus_OK,
    messageAs: coordinates
    )

    self.commandDelegate!.send(
      pluginResult,
      callbackId: command.callbackId
    )
  }
}
